packages/devtools_app/lib/src/screens/memory/panes/leaks/primitives/analysis_status.dart

```diff
@@ -32,15 +39,32 @@ class AnalysisStatusController {
     status.value = AnalysisStatus.NotStarted;
     message.value = '';
   }
+
+  void _statusChanged() async {
+    if (status.value == AnalysisStatus.ShowingResult) {
+      await Future.delayed(_Constants.showingResultDelay);
+      reset();
+    }
+  }
+
+  void _messageChanged() async {
+    await Future.delayed(_Constants.delayForUiToHandleState);
```

Comment:
I'm skeptical that these delays are needed like this. We should be able to split up whatever expensive operation we have into batches of work. It is a bit of code smell that we are adding a delay so that some event has hopefully completed but without guarantee that it has. Batch processing will use a small delay in the middle of processing data to release the UI thread periodically, and that is okay, but awaiting some magic value here doesn't feel like the right way to handle these state changes.

If you search for 'delayForBatchProcessing' in DevTools, you'll see several places where we split up an expensive data processing operation into batches.